sources/parsers/parsers.ml

```diff
@@ -47,37 +47,63 @@ let register_parser ~lang new_parser =
     end;
   parsers := (lang, new_parser) :: !parsers
 
+let debug_no_parser_for_no_extension () =
+  if verbose() then
+    eprintf
+      "error: no extension found, can't infer input format@."
+
 let debug_no_parser_for_extension s =
   if verbose() then
     eprintf
-      "error: no parser registered for ext %S. Using default lang ...@." s
+      "error: no parser registered for ext %S @." s
 
 let debug_no_parser_for_input_format s =
   if verbose() then
     eprintf
       "error: no parser registered for the provided input format %S ?@." s
 
+let get_input_parser () =
+  match Options.input_format () with
+  | Options.Native -> begin
+      try List.assoc ".ae" !parsers
+      with Not_found ->
+        debug_no_parser_for_input_format "ae";
+        exit 1
+    end
+  | Options.Smtlib2 -> begin
+      try List.assoc ".smt2" !parsers
+      with Not_found ->
+        debug_no_parser_for_input_format "smtlib";
+        exit 1
+    end
+  | Options.Why3 -> begin
+      try List.assoc ".why3" !parsers
+      with Not_found ->
+        debug_no_parser_for_input_format "why3";
+        exit 1
+    end
```

Comment:
There is basically the same code repeated thrice here, you should probably abstract that into a function of its own and call it instead of having copies of the same code, ^^